### PR TITLE
[4.0] Initial form fields namespaces

### PR DESCRIPTION
**Pull request by fastslack**
### Summary of Changes

This PR pretend to replace old fields classes to namespaces

### Testing Instructions

Install and run Joomla! normaly without issues in forms

### Expected result

All work as usual

### CODE REVIEW

File: libraries/src/Joomla/CMS/Form/Field/CalendarField.php [hunk 1]
```diff
@@ -190,8 +194,8 @@ public function setup(SimpleXMLElement $element, $value, $group = null)
 	 */
 	protected function getInput()
 	{
-		$config    = JFactory::getConfig();
-		$user      = JFactory::getUser();
+		$config    = Factory::getConfig();
```
Comment (C-Lodder) [inline libraries/src/Joomla/CMS/Form/Field/CalendarField.php]: whitespace here

File: libraries/src/Joomla/CMS/Form/Field/CalendarField.php [hunk 2]
```diff
@@ -190,8 +194,8 @@ public function setup(SimpleXMLElement $element, $value, $group = null)
...
+		$user      = Factory::getUser();
```
Comment (C-Lodder) [inline libraries/src/Joomla/CMS/Form/Field/CalendarField.php]: and here

**Comment by laoneo** [comment]
According to the discussion in Krakow with @mbabker we can't namespace the fields in libraries/joomla in the 3 series, because of autoloading. So this pr should be targeted against the 4.0-dev branch.

**Comment by mbabker** [comment]
The two in `libraries/legacy` are fine to do for 3.8.  But the rest in `libraries/joomla` have to wait for 4.0 since the directory they are in now is not autoloaded and this move would make the classes autoloaded and we've already run into problems with duplicated classes from the core field offerings.

**Comment by fastslack** [comment]
@laoneo @mbabker Im targeting this to 4.0 and create a new PR with the legacy fields

**Comment by fastslack** [comment]
@laoneo @mbabker Should I close this and made 2 PR again?

**Comment by laoneo** [comment]
You can change the base branch of the pr. What you need to do then is to merge it with the 4.0-dev branch.

**Comment by fastslack** [comment]
Hmm better to close this and open 2 PR separately

**Pull request closed without merging**